### PR TITLE
Add llama decoration support

### PR DESCRIPTION
**Pull request by DoctorMacc**

### CODE REVIEW

File: connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/LlamaEntity.java [hunk 1]
```diff
@@ -47,6 +47,25 @@ public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession s
...
+        if (entityMetadata.getId() == 20) {
+            // Bedrock treats llama decoration as armor
+            MobArmorEquipmentPacket equipmentPacket = new MobArmorEquipmentPacket();
+            equipmentPacket.setRuntimeEntityId(getGeyserId());
+            // -1 means no armor
+            if ((int) entityMetadata.getValue() != -1) {
+                // The damage value is the dye color that Java sends us
+                // Always going to be a carpet so we can hardcode 171
+                // The int then short conversion is required or we get a ClassCastException
+                equipmentPacket.setChestplate(ItemData.of(171, (short)((int) entityMetadata.getValue()), 1));
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/LlamaEntity.java:60]: The carpet ID should be constant inside of `BlockTranslator`.

File: connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/LlamaEntity.java [hunk 2]
```diff
@@ -47,6 +47,25 @@ public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession s
...
+            } else equipmentPacket.setChestplate(ItemData.of(0, (short) 0, 0));
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/LlamaEntity.java:61]:
```java
} else {
    equipmentPacket.setChestplate(ItemData.of(0, (short) 0, 0));
}
```

Comment (SupremeMortal) [reply connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/LlamaEntity.java:61]: Also, I'm pretty sure I created `ItemData.AIR`

Comment (DoctorMacc) [reply connector/src/main/java/org/geysermc/connector/entity/living/animal/horse/LlamaEntity.java:61]: Yes, I noticed that. Will implement.

File: connector/src/main/java/org/geysermc/connector/network/translators/block/BlockTranslator.java [hunk 3]
```diff
@@ -55,6 +55,9 @@
     private static final Int2ObjectMap<BlockState> BEDROCK_TO_JAVA_BLOCK_MAP = new Int2ObjectOpenHashMap<>();
     private static final IntSet WATERLOGGED = new IntOpenHashSet();
 
+    // Carpet ID, used in LlamaEntity.java
+    public static final int LLAMA_ARMOR_ID = 171;
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/network/translators/block/BlockTranslator.java:59]: Just call this `CARPET` as it is the ID for the carpet block.

**Pull request merged**